### DISCUSSION TITLE
Encoding the * parameter in query params

### DISCUSSION THREAD

**Discussion by NathanHazout** [sindresorhus/got#1524] [Q&A] [answered]
I use `URLSearchParams` to pass query parameters to `got`.

The string `$.phoneNumber[*]` gets encoded into `%24.phoneNumber%5B*%5D`, which my swagger API rejects.  
Notice the `*` in the middle? It wasn't encoded. 

Meanwhile, the module `query-string` encodes the same string to `%24.phoneNumber%5B%2A%5D`, which gets accepted by swagger.

Not sure what to do here.

**Comment by szmarczak** [comment, marked as answer]
This is the correct behaviour according to the WHATWG URL spec.

https://url.spec.whatwg.org/#url-code-points

You can pass non-compliant query string by providing the URL as a string.

**Comment by Giotino** [comment]
`query-strings` use `strict-uri-encode` which follows RFC 3986 (`strict-uri-encode` assumes that all the [reserved characters](https://tools.ietf.org/html/rfc3986#section-2.2) are delimiters and encode them)

`URLSearchParams` follows `WHATWG`

These are 2 different standards, probably your server is not compatible with `WHATWG`

**Comment by NathanHazout** [comment]
If I want to keep using `got`, what are my options?

I guess I can manually build the URL, I want to know if there are other options to keep it clean.

**Comment by szmarczak** [comment]
> You can pass non-compliant query string by providing the URL as a string.
